Core/src/EnvisionException.cpp

```diff
@@ -39,13 +39,12 @@ bool& EnvisionException::assertOnThrow()
 }
 
 EnvisionException::EnvisionException()
-{
-	if (assertOnThrow()) Q_ASSERT_X(false, "Exception thrown", "");
-}
+	: EnvisionException("")
+{}
 
-EnvisionException::EnvisionException(const QString& message) : msg_{message}
+EnvisionException::EnvisionException(const QString& message, bool throwInDebugMode) : msg_{message}
```

Comment:
This parameter is not named properly. If it is set to true, the name suggests that the exception will be thrown, but what we get is the old behavior where there is an assertion violation.

Call this `doNotAssertInDebugMode` and make it false by default.